util/src/main/java/tc/oc/pgm/util/text/types/PlayerComponent.java

```diff
@@ -81,6 +68,109 @@ static Component of(CommandSender sender, NameStyle style) {
         : TranslatableComponent.of("misc.console", TextColor.DARK_AQUA);
   }
 
+  // What an offline or vanished username renders as
+  static Component formatOffline(String name, boolean plain) {
```

Comment:
Make all of these functions return `Component.Builder` to avoid copying `Component`s among the methods. In `PlayerComponent.of` you can build it before returning it.

Reply:
Ah better idea! I’ll fix this up 👍

Reply:
I believe you meant `TextComponent.Builder`, but it’s all cleaned up now 👍